sdk/monitor/azure-monitor-ingestion/src/main/java/com/azure/monitor/ingestion/LogsIngestionAsyncClient.java

```diff
@@ -182,157 +182,134 @@ public Mono<Response<Void>> uploadWithResponse(
      */
-    private ResponseError mapToResponseError(HttpResponseException ex) {
-        ResponseError responseError = null;
-        // with DPG clients, the error object is just a LinkedHashMap and should map to the standard error
-        // response structure
-        // https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/data-plane/v1/types.json#L46
-        if (ex.getValue() instanceof LinkedHashMap<?, ?>) {
-            @SuppressWarnings("unchecked")
-            LinkedHashMap<String, Object> errorMap = (LinkedHashMap<String, Object>) ex.getValue();
-            if (errorMap.containsKey("error")) {
-                Object error = errorMap.get("error");
-                if (error instanceof LinkedHashMap<?, ?>) {
-                    @SuppressWarnings("unchecked")
-                    LinkedHashMap<String, String> errorDetails = (LinkedHashMap<String, String>) error;
-                    if (errorDetails.containsKey("code") && errorDetails.containsKey("message")) {
-                        responseError = new ResponseError(errorDetails.get("code"), errorDetails.get("message"));
-                    }
-                }
-            }
-        }
-        return responseError;
-    }
-
-    private UploadLogsResult createResponse(List<UploadLogsResult> results) {
-        int failureCount = 0;
-        List<UploadLogsError> errors =  new ArrayList<>();
-        for (UploadLogsResult result : results) {
-            if (result.getStatus() != UploadLogsStatus.SUCCESS) {
-                failureCount++;
-                errors.addAll(result.getErrors());
-            }
-        }
-        if (failureCount == 0) {
-            return new UploadLogsResult(UploadLogsStatus.SUCCESS, errors);
-        }
-        if (failureCount < results.size()) {
-            return new UploadLogsResult(UploadLogsStatus.PARTIAL_FAILURE, errors);
-        }
-        return new UploadLogsResult(UploadLogsStatus.FAILURE, errors);
-    }
-
-    private List<byte[]> createGzipRequests(List<Object> logs, ObjectSerializer serializer,
-                                            List<List<Object>> logBatches) {
+    private void createHttpRequest(ObjectSerializer serializer, Iterator<Object> iterator, FluxSink<LogsIngestionRequest> emitter) {
         try {
-            List<byte[]> requests = new ArrayList<>();
             long currentBatchSize = 0;
             ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-
             JsonGenerator generator = JsonFactory.builder().build().createGenerator(byteArrayOutputStream);
             generator.writeStartArray();
             List<String> serializedLogs = new ArrayList<>();
-
-            int currentBatchStart = 0;
-            for (int i = 0; i < logs.size(); i++) {
-                byte[] bytes = serializer.serializeToBytes(logs.get(i));
+            List<Object> originalLogsRequest = new ArrayList<>();
+            while (iterator.hasNext()) {
```

Comment:
is `flux.create` guarantees this callback will run sequentially even when concurrency is > 1? Otherwise, iterator looks dangerous here

Reply:
Yes, the concurrency only impacts the downstream call `flatMapSequential`. `Flux.create()` will run sequentially.